Source/Scenes/Overworld.cs

```diff
@@ -375,100 +447,117 @@ public override void Render(Target target)
 			batch.Clear();
 		}
 
 		// draw each entry to the screen
 		var camera = new Camera
 		{
 			Target = target,
-			Position = new Vec3(0, -100 + 30 * Ease.Cube.In(cameraCloseUpEase), 0),
+			Position = new Vec3(0, -125 + 30 * Ease.Cube.In(cameraCloseUpEase), 0),
 			LookAt = new Vec3(0, 0, 0),
 			NearPlane = 1,
 			FarPlane = 1000
 		};
 
-		if (Paused && pauseMenu != null)
+		for (int i = 0; i < entries.Count; i++)
 		{
-			pauseMenu.Update();
+			var it = entries[i];
+			var shift = Ease.Cube.In(1.0f - it.HighlightEase) * 30 - Ease.Cube.In(it.SelectionEase) * 30;
+			if (i != index)
+				shift += Ease.Cube.InOut(selectedEase) * 50;
+			var position = new Vec3((i - slide) * 60, shift, 0);
+			var rotation = Ease.Cube.InOut(it.SelectionEase);
+			var matrix =
+				Matrix.CreateScale(new Vec3(it.SelectionEase >= 0.50f ? -1 : 1, 1, 1)) *
+				Matrix.CreateRotationX(wobble.Y * it.HighlightEase) *
+				Matrix.CreateRotationZ(wobble.X * it.HighlightEase) *
+				Matrix.CreateRotationZ((state == States.Entering ? -1 : 1) * rotation * MathF.PI) *
+				Matrix.CreateTranslation(position);
+
+			if (material.Shader?.Has("u_matrix") ?? false)
+				material.Set("u_matrix", matrix * camera.ViewProjection);
+			if (material.Shader?.Has("u_near") ?? false)
+				material.Set("u_near", camera.NearPlane);
+			if (material.Shader?.Has("u_far") ?? false)
+				material.Set("u_far", camera.FarPlane);
+			if (material.Shader?.Has("u_texture") ?? false)
+				material.Set("u_texture", it.Target);
+			if (material.Shader?.Has("u_texture_sampler") ?? false)
+				material.Set("u_texture_sampler", new TextureSampler(TextureFilter.Linear, TextureWrap.ClampToEdge, TextureWrap.ClampToEdge));
+
+			var cmd = new DrawCommand(target, mesh, material)
+			{
+				DepthMask = true,
+				DepthCompare = DepthCompare.Less,
+				CullMode = CullMode.None
+			};
+			cmd.Submit();
+		}
+
+		// mod icons
+		if (selectedModIdx > 0) // don't render if not filtering
+		{
+			for (int i = 0; i < modsWithLevels.Count; i++)
+			{
+				if (i == 0) continue; // don't render the vanilla "mod"
 
-			var bounds = new Rect(0, 0, target.Width, target.Height);
+				GameMod mod = modsWithLevels[i];
 
-			pauseMenu.Render(batch, bounds.Center);
-			batch.Render(target);
-			batch.Clear();
+				bool sel = i == selectedModIdx;
+				int relativeIndex = i - selectedModIdx;
+
+				var modIcon = mod.Subtextures.TryGetValue(mod.ModInfo.Icon ?? "", out var value) ? value : strawberryImage;
+				var modIconSelectedSize = sel ? ModIconSizeLarge : ModIconSize;
+				var modIconSize = new Vec2(modIconSelectedSize / modIcon.Width, modIconSelectedSize / modIcon.Height);
+
+				batch.Image(
+					modIcon,
+					new Vec2(
+						(sel ? -(ModIconSizeLarge - ModIconSize) : 0) + ModIconLeftMargin, // Horizontal
+						(sel ? -(ModIconSizeLarge - ModIconSize) : 0) + (ModIconSpacing * relativeIndex) + (bounds.Height / 2) - ModIconVertAdjust // Vertical
+					),
+					Vec2.Zero, modIconSize, 0, Color.White);
+			}
 		}
```

Comment:
Not exactly a code issue, but when I tested this myself, I noticed if you had enough level mods, sometimes the mod icons would go behind the fuji version number labels.
Not a blocking issue, because I don't really know how to fix this without reworking the entire layout of this screen, but maybe this could be something we look into if we ever revisit in a future commit.